### ISSUE TITLE
import_from_git not working with plugins

### ISSUE THREAD

**Issue by NiklasWebnuts** [fastlane/fastlane#11112]
🔑

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When running import_from_git to load a Fastfile from another repository, none of the dependencies (e.g. fastlane plugins) are installed. For me, this is extremely unintuitive and ruins the otherwise lovely abstraction that import_from_git provides.

I have the following setup:
- **Repository A** - containing information that uses lanes from **Repository B**
- **Repository B** - Containing lanes using the plugin [upgrade_super_old_xcode_project](https://github.com/ionic-zone/fastlane-plugin-upgrade_super_old_xcode_project)

Fastfile in **Repository A**: 

```
fastlane_version "2.64.0"

default_platform :ios

import_from_git(url: 'git@github.com:myorg/fastlane-actions.git')

before_all do
  # Name of the project. Used to find the Xcode project
  ENV["PROJECT_NAME"] = "Sales Tool"

  # Project directories (used for various Fastlane actions)
  ENV["PROJECT_DIRECTORY"] = "cordova/platforms/ios/#{ENV['PROJECT_NAME']}.xcodeproj"
  ENV["EMBER_PROJECT_DIRECTORY"] = "../sales-app"
  ENV["CORDOVA_PROJECT_DIRECTORY"] = "../cordova"
end

lane :test do
  ensure_correct_xcode_project(
    path: ENV["PROJECT_DIRECTORY"]
  )
end
```

Fastfile in **Repository B**:

```
fastlane_version "2.68.0"

lane :ensure_correct_xcode_project do |options|
  # Ensure that the user has specified the xcode project location
  unless options.key?(:path)
    UI.user_error!("'path' option not provided.")
  end

  # Upgrade the Xcode files to be compatible with fastlane
  # Older versions (generated by Cordova) are missing some files, which are required by fastlane
  upgrade_super_old_xcode_project(
    path: options[:path],
    team_id: CredentialsManager::AppfileConfig.try_fetch_value(:team_id)
  )
end
```

Pluginfile in **Repository B**:

```
gem 'fastlane-plugin-upgrade_super_old_xcode_project'
```

The required plugin is installed correctly in **Repository B** and the lane `ensure_correct_xcode_project` is working when used without/outside `import_from_git`.

Are there any workarounds for this issue that does not include installing the plugin in **Repository A**?

### Captured Output

Command Used: `test`
<details><summary>Output/Log</summary>

```

DEBUG [2017-12-05 15:45:13.71]: Checking if there are any plugins that should be loaded...
DEBUG [2017-12-05 15:45:13.71]: Loading 'fastlane-plugin-ftp' plugin
INFO [2017-12-05 15:45:13.77]: gem 'fastlane-plugin-ftp' is already installed
+---------------------+---------+--------+
|              Used plugins              |
+---------------------+---------+--------+
| Plugin              | Version | Action |
+---------------------+---------+--------+
| fastlane-plugin-ftp | 0.1.3   | ftp    |
+---------------------+---------+--------+

Successfully loaded Appfile at path '/Users/niklas/Documents/git/project/fastlane/Appfile'
- app_identifier: 'com.project.salestool'
- apple_id: 'myid@example.com'
- team_id: 'someteam'
-------
INFO [2017-12-05 15:45:14.02]: -------------------------------------------------
INFO [2017-12-05 15:45:14.02]: --- Step: Verifying required fastlane version ---
INFO [2017-12-05 15:45:14.02]: -------------------------------------------------
INFO [2017-12-05 15:45:14.02]: Your fastlane version 2.68.1 matches the minimum requirement of 2.64.0  ✅
INFO [2017-12-05 15:45:14.06]: ------------------------------
INFO [2017-12-05 15:45:14.06]: --- Step: default_platform ---
INFO [2017-12-05 15:45:14.06]: ------------------------------
INFO [2017-12-05 15:45:14.07]: -----------------------------
INFO [2017-12-05 15:45:14.07]: --- Step: import_from_git ---
INFO [2017-12-05 15:45:14.07]: -----------------------------
INFO [2017-12-05 15:45:14.10]: Cloning remote git repo...
INFO [2017-12-05 15:45:14.10]: $ GIT_TERMINAL_PROMPT=0 git clone 'git@github.com:myorg/fastlane-actions.git' '/var/folders/jd/nl_3vr_n3_n0w7njmkkgvkbh0000gn/T/fl_clone20171205-18862-1rj07th/fastlane-actions.git' --depth 1 -n
INFO [2017-12-05 15:45:14.12]: ▸ Cloning into '/var/folders/jd/nl_3vr_n3_n0w7njmkkgvkbh0000gn/T/fl_clone20171205-18862-1rj07th/fastlane-actions.git'...
INFO [2017-12-05 15:45:15.81]: $ cd '/var/folders/jd/nl_3vr_n3_n0w7njmkkgvkbh0000gn/T/fl_clone20171205-18862-1rj07th/fastlane-actions.git' && git checkout HEAD 'fastlane/Fastfile'
INFO [2017-12-05 15:45:15.83]: $ cd '/var/folders/jd/nl_3vr_n3_n0w7njmkkgvkbh0000gn/T/fl_clone20171205-18862-1rj07th/fastlane-actions.git' && git checkout HEAD 'fastlane/actions'
INFO [2017-12-05 15:45:15.84]: ▸ error: pathspec 'fastlane/actions' did not match any file(s) known to git.
INFO [2017-12-05 15:45:15.95]: gem 'commonjs' is already installed
INFO [2017-12-05 15:45:16.05]: gem 'therubyracer' is already installed
INFO [2017-12-05 15:45:16.11]: -------------------------------------------------
INFO [2017-12-05 15:45:16.11]: --- Step: Verifying required fastlane version ---
INFO [2017-12-05 15:45:16.11]: -------------------------------------------------
INFO [2017-12-05 15:45:16.11]: Your fastlane version 2.68.1 matches the minimum requirement of 2.68.0  ✅
INFO [2017-12-05 15:45:16.12]: Driving the lane 'test' 🚀
INFO [2017-12-05 15:45:16.13]: ---------------------------------------------------------
INFO [2017-12-05 15:45:16.13]: --- Step: Switch to ensure_correct_xcode_project lane ---
INFO [2017-12-05 15:45:16.13]: ---------------------------------------------------------
INFO [2017-12-05 15:45:16.13]: Cruising over to lane 'ensure_correct_xcode_project' 🚖
DEBUG [2017-12-05 15:45:16.13]: /Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:155:in `trigger_action_by_name'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
../../../../../../var/folders/jd/nl_3vr_n3_n0w7njmkkgvkbh0000gn/T/fl_clone20171205-18862-1rj07th/fastlane-actions.git/fastlane/Fastfile:73:in `block in parsing_binding'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/lane.rb:33:in `call'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:201:in `try_switch_to_lane'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:156:in `trigger_action_by_name'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
Fastfile:18:in `block in parsing_binding'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/lane.rb:33:in `call'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/lane_manager.rb:54:in `cruise_lane'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
/Users/niklas/.rvm/gems/ruby-2.4.2/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
/Users/niklas/.rvm/gems/ruby-2.4.2/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
/Users/niklas/.rvm/gems/ruby-2.4.2/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
/Users/niklas/.rvm/gems/ruby-2.4.2/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/commands_generator.rb:304:in `run'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/fastlane/lib/fastlane/cli_tools_distributor.rb:75:in `take_off'
/Users/niklas/.rvm/gems/ruby-2.4.2@global/gems/fastlane-2.68.1/bin/fastlane:20:in `<top (required)>'
/Users/niklas/.rvm/gems/ruby-2.4.2/bin/fastlane:23:in `load'
/Users/niklas/.rvm/gems/ruby-2.4.2/bin/fastlane:23:in `<main>'
/Users/niklas/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
/Users/niklas/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
WARN [2017-12-05 15:45:16.13]: Lane Context:
INFO [2017-12-05 15:45:16.13]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>nil, :LANE_NAME=>"test"}
ERROR [2017-12-05 15:45:16.13]: Could not find action, lane or variable 'upgrade_super_old_xcode_project'. Check out the documentation for more details: https://docs.fastlane.tools/actions
INFO [2017-12-05 15:45:16.13]: Successfully generated documentation at path '/Users/niklas/Documents/git/project/fastlane/README.md'

+------+---------------------------------------------+-------------+
|                         fastlane summary                         |
+------+---------------------------------------------+-------------+
| Step | Action                                      | Time (in s) |
+------+---------------------------------------------+-------------+
| 1    | Verifying required fastlane version         | 0           |
| 2    | default_platform                            | 0           |
| 3    | Verifying required fastlane version         | 0           |
| 4    | import_from_git                             | 2           |
| 5    | Switch to ensure_correct_xcode_project lane | 0           |
+------+---------------------------------------------+-------------+

DEBUG [2017-12-05 15:45:16.14]: All plugins are up to date
ERROR [2017-12-05 15:45:16.14]: fastlane finished with errors
DEBUG [2017-12-05 15:45:17.46]: All plugins are up to date


```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.13.6 (Apple Git-96)           |
| Installation Source         | ~/.rvm/gems/ruby-2.4.2/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G1036)                  |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.2/lib                |
| OpenSSL Version             | OpenSSL 1.1.0g  2 Nov 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             | 🚫 |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.64.0"

default_platform :ios

import_from_git(url: 'git@github.com:myorg/fastlane-actions.git')

before_all do
  # Name of the project. Used to find the Xcode project
  ENV["PROJECT_NAME"] = "Sales Tool"

  # Project directories (used for various Fastlane actions)
  ENV["PROJECT_DIRECTORY"] = "cordova/platforms/ios/#{ENV['PROJECT_NAME']}.xcodeproj"
  ENV["EMBER_PROJECT_DIRECTORY"] = "../sales-app"
  ENV["CORDOVA_PROJECT_DIRECTORY"] = "../cordova"
end

lane :test do
  ensure_correct_xcode_project(
    path: ENV["PROJECT_DIRECTORY"]
  )
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.gludmarstrand.salestool" # The bundle identifier of your app
apple_id "myid@example.com" # Your Apple email address

team_id "someteam" # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.68.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin              | Version | Update-Status |
| ------------------- | ------- | ------------- |
| fastlane-plugin-ftp | 0.1.3   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.16.0       |
| io-console                | 0.4.6        |
| CFPropertyList            | 2.3.5        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| commonjs                  | 0.2.7        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.59.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.13.1       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.2       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.3.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-ftp       | 0.1.3        |
| libv8                     | 3.16.14.19   |
| ref                       | 2.0.0        |
| therubyracer              | 0.12.3       |
</details>

*generated on:* **2017-12-05**
</details>

**Comment by ohayon** [comment]
Hey @NiklasWebnuts, apologies for the issue you are seeing, I have two thoughts:

1. This is a pretty cool use case that you have here.
2. We should definitely make sure this works. 

We will queue up a fix for this, but in the meantime is there any chance you might want to submit a PR and beat us to the punch here? 🚀  🐙

**Comment by NiklasWebnuts** [comment]
Update: I realised that what I'm really trying to do is define a set of actions to use throughout my different projects. Even though the solution above would work (i.e. switching lanes a bunch of times), I've decided to try a plugin-approach instead. 

In my plugin I've defined an action `ensure_correct_xcode_project`, in which I load the and use the _upgrade_super_old_xcode_project_ plugin like so:

```ruby
def self.run(params)
    FastlaneRequire.install_gem_if_needed(gem_name: 'fastlane-plugin-upgrade_super_old_xcode_project', require_gem: true)
    Fastlane::Actions::UpgradeSuperOldXcodeProjectAction.run(
        path: params[:path],
        team_id: params[:team_id]
    )
end
```

This seems to work without having to install the _upgrade_super_old_xcode_project_ plugin in **Repository A**. However, I feel very dirty doing something like this. 

I'm sorry that this is a bit off topic in regards to the original issue, however, I believe that it might be interesting for future visitors/readers.
